llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -1803,7 +1811,7 @@ bool VectorCombine::foldShuffleToIdentity(Instruction &I) {
           Value *V = IL.first;
           return !V || V == FrontV;
         })) {
-      SplatLeafs.insert(FrontV);
+      ConstLeafs.insert(FrontV);
```

Comment:
Isn't this an ugly special case? Can't you handle constant splats in the cost code?